packages/permission_handler/tizen/src/permission_manager.cc

```diff
@@ -108,16 +110,83 @@ void PermissionManager::RequestPermissions(
   }
 
   on_going_ = true;
-  request_success_callback_ = success_callback;
-  request_error_callback_ = error_callback;
-  result = ppm_request_permissions(permissions_to_request.data(),
-                                   permissions_to_request.size(),
-                                   OnRequestPermissionsResponse, this);
-  if (result != PRIVACY_PRIVILEGE_MANAGER_ERROR_NONE) {
+
+  struct Param {
+    PermissionManager* self{nullptr};
+    bool is_done{false};
+    size_t remaining_request{0};
+  };
+  Param p;
+  p.self = this;
+  p.remaining_request = permissions_to_request.size();
+
+  for (size_t i = 0; i < permissions_to_request.size(); i++) {
+    const char* permission = permissions_to_request[i];
+    p.is_done = false;
+    result = ppm_request_permission(
```

Comment:
how about extracting this callback method?

Reply:
Well... this is close to my personal preference,  I like to avoid adding unnecessary functions to class members or exposing them in headers and I generally prefer lambda functions unless they are used forever elsewhere. 
However, I respect your opinion and it is possible :)  Should I change it?

Reply:
😃  this is also my preference, not a critical. I just found this lambda function too long to read.

Reply:
I've updated this PR